### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by romans-adi**
For this pull request:

The app was created based on project requirements.

# Logic, Redux, API

- Added API request functions for fetching data from the external server.
- Created Redux `actions` and `async thunks` for fetching data.
- Separated the logic for fetching meta data from the main data retrieval process, enabling the retrieval of `meta` data separately and efficiently.
- Incorporated `axios-rate-limiter` to limit the number of API requests made within a specific timeframe, ensuring optimal usage of API resources and preventing abuse or excessive usage.
- Actively used `dispatchers` to manage state updates and trigger actions in the Redux store.
- Utilized the `useEffect` hook to control the flow of the application, including fetching data, updating state, and handling side effects.
- Employed dispatchers to trigger actions based on user interactions, such as fetching data on page load, updating state based on user input, or triggering specific functionality in response to user actions.

# Layout & Styling
- Implemented components for rendering data on different pages.
- Created a `navbar` component with navigation and page-specific functionality.
- Applied styling updates based on the provided design.

# Testing

- Added unit tests for all the units.
- Utilized `'jest-mock-axios` for mocking API requests and responses, ensuring independent and controlled testing of the application.
- Created a store mock using `redux-mock-store`, allowing for isolated testing of Redux actions and reducers.

# Routing

- Implemented routing functionality using a library `React Router`, allowing for the navigation between different pages within the application.
- Configured different paths for each page, ensuring that the appropriate components are rendered based on the URL.
- Integrated a "back" button functionality, enabling users to navigate back to the previous page or step in the application flow.
- Verified the functionality of the "back" button to ensure it works properly and correctly takes users to the intended previous page.
***
_Please note that the API has a request limit, and it is possible to accidentally exceed the limit. If you encounter any issues due to receiving a `429 error` (Too Many Requests), please contact me. I would also let you know that this PR includes disabled linting rules for unpredictable errors (SCSS @use, ESlint extraneous dependencies)._

### CODE REVIEW

File: src/__tests__/CategoryItem.test.js
Comment (joskalenda) [inline src/__tests__/CategoryItem.test.js:1]: - [ ] Hey dear football lover 😄, you did a great job, and your api Has a great topy though. however, some of your test cases are falling, as we mentioned on Slack, please them and request another review

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @romans-adi , |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
|  |

## Highlights

- [ ].. ✅ Green linters
- [ ].. ✅ Great design
- [ ].. ✅ Great readme file

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/CategoryItem.test.js
Comment (romans-adi) [reply src/__tests__/CategoryItem.test.js:1]: Thank you for your feedback, @joskalenda!
I wish you a good weekend :champagne:

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @romans-adi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ You tested most of your components, very good job. 👍🏾 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by romans-adi** [comment]
Thank you very much for your feedback, @bobb-Rob! 
However I found one major issue after sent my project on review :smiley: But I promise that I will fix it as soon as possible!
Have a good day! :champagne:

**Pull request merged**